data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -22,8 +24,16 @@
 
 class Utils {
 
-    static Results post(String url, String aadAccessToken, String payload) throws DataServiceException, DataClientException {
-        HttpClient httpClient = HttpClients.createSystem();
+    static Results post(String url, String aadAccessToken, String payload, Long timeoutMs) throws DataServiceException, DataClientException {
+
+        HttpClient httpClient;
+        if (timeoutMs != null) {
+            RequestConfig requestConfig = RequestConfig.custom().setSocketTimeout(timeoutMs.intValue()).build();
```

Comment:
I get that the socket timeout doesn't allow longs, so maybe we just use int from the start?

Reply:
problem is TimeUnit is long, so I either do the conversion here, or on the caller. 
would you prefer it in the caller?